### PR TITLE
Fix #457: Refactor javax.annotation to ChangePackage

### LINKED ISSUE #457: `@PostConstruct` import being removed from beans
## What version of OpenRewrite are you using?

**Latest**

## How are you running OpenRewrite?

CLI using `mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-migrate-java:RELEASE -Drewrite.activeRecipes=org.openrewrite.java.migrate.jakarta.JakartaEE10`

CDI Spec: https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.pdf

Before conversion:

```java
package org.primefaces.showcase.view.overlay;

import org.primefaces.showcase.domain.Movie;

import javax.annotation.PostConstruct;
import javax.enterprise.context.RequestScoped;
import javax.inject.Named;
import java.util.ArrayList;
import java.util.List;

import javax.faces.application.FacesMessage;
import javax.faces.context.FacesContext;

import org.primefaces.event.SelectEvent;

@Named
@RequestScoped
public class MovieView {

    private Movie movie;

    private List<Movie> movieList;

    public List<Movie> getMovieList() {
        return movieList;
    }

    @PostConstruct
    public void init() {
        movieList = new ArrayList<Movie>();

        movieList.add(new Movie("The Lord of the Rings: The Two Towers", "Peter Jackson", "Fantasy, Epic", 179));
        movieList.add(new Movie("The Amazing Spider-Man 2", "Marc Webb", "Action", 142));
        movieList.add(new Movie("Iron Man 3", "Shane Black", "Action", 109));
        movieList.add(new Movie("Thor: The Dark World", "Alan Taylor", "Action, Fantasy", 112));
        movieList.add(new Movie("Avatar", "James Cameron", "Science Fiction", 160));
        movieList.add(new Movie("The Lord of the Rings: The Fellowship of the Ring", "Peter Jackson", "Fantasy, Epic", 165));
        movieList.add(new Movie("Divergent", "Neil Burger", "Action", 140));
        movieList.add(new Movie("The Hobbit: The Desolation of Smaug", "Peter Jackson", "Fantasy", 161));
        movieList.add(new Movie("Rio 2", "Carlos Saldanha", "Comedy", 101));
        movieList.add(new Movie("Captain America: The Winter Soldier", "Joe Russo", "Action", 136));
        movieList.add(new Movie("Fast Five", "Justin Lin", "Action", 132));
        movieList.add(new Movie("The Lord of the Rings: The Return of the King", "Peter Jackson", "Fantasy, Epic", 200));

    }

    public Movie getMovie() {
        return movie;
    }

    public void setMovie(Movie movie) {
        this.movie = movie;
    }

    public void onRowSelect(SelectEvent<Movie> event) {
        FacesMessage msg = new FacesMessage("Movie Selected", event.getObject().getMovie());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }
}
```

After Conversion:

```java
package org.primefaces.showcase.view.overlay;

import org.primefaces.showcase.domain.Movie;

import jakarta.enterprise.context.RequestScoped;
import jakarta.inject.Named;
import java.util.ArrayList;
import java.util.List;

import jakarta.faces.application.FacesMessage;
import jakarta.faces.context.FacesContext;

import org.primefaces.event.SelectEvent;

@Named
@RequestScoped
public class MovieView {

    private Movie movie;

    private List<Movie> movieList;

    public List<Movie> getMovieList() {
        return movieList;
    }

    @PostConstruct
    public void init() {
        movieList = new ArrayList<Movie>();

        movieList.add(new Movie("The Lord of the Rings: The Two Towers", "Peter Jackson", "Fantasy, Epic", 179));
        movieList.add(new Movie("The Amazing Spider-Man 2", "Marc Webb", "Action", 142));
        movieList.add(new Movie("Iron Man 3", "Shane Black", "Action", 109));
        movieList.add(new Movie("Thor: The Dark World", "Alan Taylor", "Action, Fantasy", 112));
        movieList.add(new Movie("Avatar", "James Cameron", "Science Fiction", 160));
        movieList.add(new Movie("The Lord of the Rings: The Fellowship of the Ring", "Peter Jackson", "Fantasy, Epic", 165));
        movieList.add(new Movie("Divergent", "Neil Burger", "Action", 140));
        movieList.add(new Movie("The Hobbit: The Desolation of Smaug", "Peter Jackson", "Fantasy", 161));
        movieList.add(new Movie("Rio 2", "Carlos Saldanha", "Comedy", 101));
        movieList.add(new Movie("Captain America: The Winter Soldier", "Joe Russo", "Action", 136));
        movieList.add(new Movie("Fast Five", "Justin Lin", "Action", 132));
        movieList.add(new Movie("The Lord of the Rings: The Return of the King", "Peter Jackson", "Fantasy, Epic", 200));

    }

    public Movie getMovie() {
        return movie;
    }

    public void setMovie(Movie movie) {
        this.movie = movie;
    }

    public void onRowSelect(SelectEvent<Movie> event) {
        FacesMessage msg = new FacesMessage("Movie Selected", event.getObject().getMovie());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }
}

```

### ERROR

Instead of converting `import javax.annotation.PostConstruct;` it just removed the `import` entirely.

### PR DESCRIPTION
**Pull request by melloware**
## What's changed?
Fix #457: Refactor javax.annotation to ChangePackage

Change from renaming individual Classes to ChangePackage.  Not sure why it was going to all this trouble for individual names when one ChangePackage will do.

## What's your motivation?
- Fixes #457

### Checklist
- [X] All existing tests pass
- [X] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by melloware** [comment]
YUCK....let me research.

**Comment by melloware** [comment]
@timtebeek i made the change but i think maybe the README.MD needs to be update with build instructions?  `gradlew build` or `gradlew install` i can't figure out how to build the snapshot in my local `/m2` repo?

**Comment by timtebeek** [comment]
> @timtebeek i made the change but i think maybe the README.MD needs to be update with build instructions? `gradlew build` or `gradlew install` i can't figure out how to build the snapshot in my local `/m2` repo?

Ah yes I can see it might take some getting used to with Gradle; the instructions are documented here:
- https://docs.openrewrite.org/authoring-recipes/recipe-development-environment#local-publishing-for-testing
- https://docs.openrewrite.org/reference/building-openrewrite-from-source#building-openrewrite

TLDR: `./gradlew publishToMavenLocal` does what `mvn install` would otherwise do.

Thanks again for providing the fix here! Great to have that troublesome behavior of ChangeType-in-bulk-on-larger-projects resolved.

**Comment by mr1chter** [comment]
Hi! not sure if I'm right here or if I should open a proper bug report. This MR kinda broke my jakarta ee 10 migration, because we use a few spotbugs annotations that also live under javax.annotations (which are part of JSR305 (mostly frozen?), we get them via com.google.code.findbugs:annotations, which gets pulled in by spotbugs-annotations). 

I think in our case those annotations could just as well get migrated to the associated spotbugs annotations (`edu.umd.cs.findbugs.annotations`) but for now they are there and while migrating them to their jakarta counterparts does not really hurt, there are a few that are not part of the jakarta namespace (because they were also not part of java ee 8), like `javax.annotation.Nonnegative`. When I ran the migration previously, those were left alone, but now they also get rewritten to `jakarta.annotation.Nonnegative` which does not compile.

I'm not sure if this use case is important enough to warrant a bug report though, since someone else exporting classes in the javax.annotation namespace is weird enough.

**Comment by melloware** [comment]
yeah that is weird they should not be doing that :) I would report a bug to SpotBugs/FindBugs to have them fix it.

**Comment by timtebeek** [comment]
Thanks for the heads up @mr1chter ; Indeed we would not have expected spotbugs to add recipes to the javax.annotation package. I'd report it there first.

For your particular case you can write your own recipe that undoes the changes made here with the addition of this step
```yaml
  - org.openrewrite.java.ChangeType:
      oldFullyQualifiedTypeName: jakarta.annotation.Nonnegative
      newFullyQualifiedTypeName: javax.annotation.Nonnegative
```

I hope that helps!

**Comment by iparadiso** [comment]
Hi @timtebeek.  I'm encountering this problem.  There are a bunch of annotations in `com.google.code.findbugs:jsr305` that are transformed.  I tried your proposed workout in a follow-on recipe, but I'm having problems with it working.  

Does org.openrewrite.java.ChangeType and org.openrewrite.java.ChangePackage require type resolution in order to work?  If so, then the following might not work since `jakarta.annotation.Nonnegative` type cannot be resolved as it doesn't exist on the classpath.

```
 - org.openrewrite.java.ChangeType:
      oldFullyQualifiedTypeName: jakarta.annotation.Nonnegative
      newFullyQualifiedTypeName: javax.annotation.Nonnegative
```

**Comment by knutwannheden** [comment]
Maybe we should add an exceptions option to ChangePackage.
